File: VelibStations_Android/common/src/androidTest/kotlin/com/muhammedsafiulazam/common/database/velib/VelibDatabaseUnitTest.kt
import com.muhammedsafiulazam.common.addon.AddOnManager
import com.muhammedsafiulazam.common.addon.AddOnType
import com.muhammedsafiulazam.common.core.AndroidUnitTest
import com.muhammedsafiulazam.common.database.IDatabaseManager
import com.muhammedsafiulazam.common.database.velib.event.VelibDatabaseEventType
import com.muhammedsafiulazam.common.event.Event
import com.muhammedsafiulazam.common.event.IEventManager
import com.muhammedsafiulazam.common.service.velib.model.Dataset
import com.muhammedsafiulazam.common.utils.CoroutineUtils
import kotlin.test.Test
import kotlin.test.asserter

class VelibDatabaseUnitTest : AndroidUnitTest() {
    @Test
    fun saveData() = CoroutineUtils.runBlocking() {
        var e: Event? = null

        val eventManager: IEventManager = AddOnManager.getAddOn(AddOnType.EVENT_MANAGER) as IEventManager
        eventManager.subscribe(callback = { event: Event -> Unit
            if (event.type.equals(VelibDatabaseEventType.SAVE_DATA)) {
                e = event
            }
        })

        val databaseManager: IDatabaseManager = AddOnManager.getAddOn(AddOnType.DATABASE_MANAGER) as IDatabaseManager
        val dataset: Dataset = Dataset(0, null, null)
        databaseManager.getVelibDatabase().saveData(dataset)

        CoroutineUtils.delay(DELAY_MINIMUM)
        asserter.assertTrue("saveData", e != e != null && e!!.data != null && e!!.error == null)
    }
}

File: VelibStations_iOS/VelibStationsTests/Common/Database/Velib/VelibDatabaseUnitTest.swift
//
//  VelibDatabaseUnitTest.swift
//  VelibStationsTests
//
//  Created by Safi on 28/11/2019.
//

import XCTest
@testable import VelibStations
@testable import CommonKit

class VelibDatabaseUnitTest : UnitTest {
    
    override func setUp() {
        // Before every test.
        super.setUp()
    }

    override func tearDown() {
        // After every test.
        super.tearDown()
    }

    func testSaveData() {
        DispatchQueue.main.async {
            var e: Event? = nil
            
            let eventManager: IEventManager = AddOnManager().getAddOn(type: AddOnType().EVENT_MANAGER) as! IEventManager
            eventManager.subscribe(callback: { event in
                e = event
            })
            
            let databaseManager: IDatabaseManager = AddOnManager().getAddOn(type: AddOnType().DATABASE_MANAGER) as! IDatabaseManager
            let dataset: Dataset = Dataset(hits: 0, parameters: nil, records: nil)
            databaseManager.getVelibDatabase().saveData(dataset: dataset)
            
            sleep(self.DELAY_MINIMUM)
            XCTAssertTrue(e != nil && e!.data != nil && e!.error != nil)
        }
    }
}

